libraries/AP_RangeFinder/AP_RangeFinder_MAVLink.cpp

```diff
@@ -51,11 +51,18 @@ void AP_RangeFinder_MAVLink::handle_msg(mavlink_message_t *msg)
     mavlink_distance_sensor_t packet;
     mavlink_msg_distance_sensor_decode(msg, &packet);
 
-    // only accept distances for downward facing sensors
-    if (packet.orientation == MAV_SENSOR_ROTATION_PITCH_270) {
-        state.last_reading_ms = AP_HAL::millis();
-        distance_cm = packet.current_distance;
+    // Check if distances from downward facing sensors
+    if (packet.orientation != MAV_SENSOR_ROTATION_PITCH_270) {
+        return;
     }
+
+    // Check if distance between the min-max range
+    if (packet.current_distance > packet.max_distance || packet.current_distance < packet.min_distance) {
+        return;
```

Comment:
Could you please brief a little bit more.
Do you mean something like this:
if packet.distace > packet.max_distance then state.max_distance_cm = packet.max_distance;
similarly for packet.min_distance.

Reply:
I mean this:

```
state.max_distance_cm = packet.max_distance;
state.min_distance_cm = packet.min_distance;
```

When `update_status` is called, it will set status to OutOfRangeHigh or OutOfRangeLow.